Fix inconsistent lag calculation in SirCAL

benjaminysmith opened the pull request:
### Description
Fix inconsistent lag calculation in SirCAL

### Changelog
- Use current date instead of metadata max date to avoid problems with metadata delays
- Add more logging to lag and date retrieval
- Fix the lag calculation by subtracting 1 (e.g., no gaps should be gap = 0 days, not gap = 1 day)

### Fixes 
- Fixes #523

chinandrew commented on sir_complainsalot/delphi_sir_complainsalot/check_source.py:
```diff
@@ -87,26 +88,36 @@ def check_source(data_source, meta, params, grace, logger):
 
         # convert numpy datetime values to pandas datetimes and then to
         # datetime.date, so we can work with timedeltas after
         unique_dates = [pd.to_datetime(val).date()
```
i know this isn't affected in this PR (though the unique issues is), but if this is unique values, should it be wrapped in `set()`?

benjaminysmith replied:
I feel like that only really matters if we were to add more values to it (to make sure that it really stayed a unique set) -- since we are using unique(), the list should still fullfill the contract of the variable name. wydt?

chinandrew replied:
oh i didnt even see the unique(), that makes sense to me. is unique() guaranteed to return them in the right order? Based on the gap_days computation line, seems like you want them monotonically increasing.

krivard replied:
unique() is order-preserving, so if they start out ordered, they'll stay that way

chinandrew replied:
:+1:

chinandrew reviewed (approved):
:shipit::shipit::shipit:

benjaminysmith commented:
Please don't merge yet -- noticed an edge case I need to fix...

benjaminysmith commented:
Added. @krivard please merge if it looks ok.

The additional check I added was for None coming back from API (since we are no longer using metadata for the last date this means for very out of date signals we might get no data back using the current date as the endpoint)

krivard reviewed (approved):
👍
